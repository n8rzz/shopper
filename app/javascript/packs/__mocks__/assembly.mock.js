export const assemblyMock = {
    id: 1,
    name: 'Souvlaki',
    created_at: '2019-07-29T22:49:47.431Z',
    updated_at: '2019-07-29T22:49:47.431Z',
};

export const assemblyListMock = [
    {
        ...assemblyMock,
    },
    {
        id: 2,
        name: 'Chilli con Carne',
        created_at: '2019-07-29T22:49:47.519Z',
        updated_at: '2019-07-29T22:49:47.519Z',
    },
];

export const assemblyMapMock = {
    1: assemblyListMock[0],
    2: assemblyListMock[1],
};
