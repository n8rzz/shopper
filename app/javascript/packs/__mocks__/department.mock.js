export const departmentListMock = [
    {
        created_at: '2019-07-21T16:03:32.883Z',
        id: 3,
        name: 'Bakery',
        updated_at: '2019-07-21T16:03:32.883Z',
    },
    {
        created_at: '2019-07-21T16:03:32.883Z',
        id: 5,
        name: 'Meat',
        updated_at: '2019-07-21T16:03:32.883Z',
    },
];

export const departmentMapMock = {
    3: departmentListMock[0],
    5: departmentListMock[1],
};
