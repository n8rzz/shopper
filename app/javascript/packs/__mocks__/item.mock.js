export const itemListMock = [
    {
        created_at: '2019-07-21T16:03:33.162Z',
        department_id: 5,
        id: 15,
        name: 'Agave Syrup',
        updated_at: '2019-07-21T16:03:33.162Z',
    },
    {
        created_at: '2019-07-21T16:03:33.162Z',
        department_id: 3,
        id: 13,
        name: 'Agave Syrup',
        updated_at: '2019-07-21T16:03:33.162Z',
    },
];

export const itemMapMock = {
    15: itemListMock[0],
    13: itemListMock[1],
};
