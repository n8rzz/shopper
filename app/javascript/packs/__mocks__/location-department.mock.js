export const locationDepartmentListMock = [
    {
        created_at: '2019-08-22T00:58:38.793Z',
        department_id: 3,
        id: 1,
        location_id: 1,
        sort_order: 0,
        updated_at: '2019-08-22T00:58:38.793Z',
    },
    {
        created_at: '2019-08-22T00:58:38.793Z',
        department_id: 5,
        id: 2,
        location_id: 1,
        sort_order: 0,
        updated_at: '2019-08-22T00:58:38.793Z',
    },
];
