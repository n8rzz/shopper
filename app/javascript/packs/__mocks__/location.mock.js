export const locationMock = {
    shoppingDate: '2019-08-27T00:00:00.000Z',
    status: 'pending',
    locationName: 'Williamson and Sons',
    locationId: 1
};
