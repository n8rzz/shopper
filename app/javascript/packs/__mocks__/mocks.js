import { assemblyItemsGroupedByAssemblyIdMock } from './assembly-item.mocks';
import { assemblyMock, assemblyListMock, assemblyMapMock } from './assembly.mock';
import { csrfMock } from './csrf.mock';
import { departmentListMock, departmentMapMock } from './department.mock';
import { itemMock, itemListMock, itemMapMock } from './item.mock';
import { locationDepartmentListMock } from './location-department.mock';
import { locationMock } from './location.mock';
import { orderItemListMock } from './order-item.mock';
import { orderMock, orderLocationNameMock } from './order.mock';

export {
    assemblyItemsGroupedByAssemblyIdMock,
    assemblyMock,
    assemblyListMock,
    assemblyMapMock,
    csrfMock,
    departmentListMock,
    departmentMapMock,
    itemMock,
    itemListMock,
    itemMapMock,
    locationDepartmentListMock,
    locationMock,
    orderItemListMock,
    orderMock,
    orderLocationNameMock,
};
