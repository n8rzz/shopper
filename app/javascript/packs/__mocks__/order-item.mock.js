export const orderItemListMock = [
    {
        id: 22,
        order_id: 11,
        item_id: 15,
        assembly_id: 1,
        created_at: '2019-07-29T22:50:37.840Z',
        updated_at: '2019-07-29T22:50:37.840Z',
        department_id: 3,
        picked: false,
        qty: 3,
    },
    {
        id: 23,
        order_id: 11,
        item_id: 13,
        assembly_id: null,
        created_at: '2019-07-29T22:50:37.887Z',
        updated_at: '2019-07-29T22:50:37.887Z',
        department_id: 5,
        picked: false,
        qty: 1,
    },
];
