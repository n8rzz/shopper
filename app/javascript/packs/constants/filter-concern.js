export const FILTER_CONCERN = {
    ALPHA: 'ALPHA',
    DEPARTMENT: 'DEPARTMENT',
    FILTER: 'FILTER',
    ITEM: 'ITEM',
    MEAL: 'MEAL',
};

export const FILTER_LABEL = {
    [FILTER_CONCERN.ALPHA]: 'Alpha',
    [FILTER_CONCERN.DEPARTMENT]: 'Department',
    [FILTER_CONCERN.FILTER]: 'Filter',
    [FILTER_CONCERN.ITEM]: 'Item',
    [FILTER_CONCERN.MEAL]: 'Meal',
};
