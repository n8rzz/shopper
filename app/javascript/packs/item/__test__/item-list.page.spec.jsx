import React from 'react';
import { shallow } from 'enzyme';

import { ItemListPage } from '../item-list.page';
import { GroupedItemList } from '../grouped-item-list.component';
import { ItemList } from '../item-list.component';
import { FILTER_CONCERN } from '../../constants/filter-concern';
import { departmentListMock, itemListMock } from '../../__mocks__/mocks';

describe('ItemListPage', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<ItemListPage
            csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
            departments={departmentListMock}
            items={itemListMock}
        />);

        expect(component).toMatchSnapshot();
    });

    describe('when `state.concern` equals `DEPARTMENT', () => {
        test('it renders a `GroupedItemList` component', () => {
            const component = shallow(<ItemListPage
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departments={departmentListMock}
                items={itemListMock}
            />);
            component.setState({ concern: FILTER_CONCERN.DEPARTMENT });

            expect(component.find(GroupedItemList).length).toBe(1);
        });
    });

    describe('when `state.concern` equals `ALPHA', () => {
        test('it renders an `ItemList` component', () => {
            const component = shallow(<ItemListPage
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departments={departmentListMock}
                items={itemListMock}
            />);
            component.setState({ concern: FILTER_CONCERN.ALPHA });

            expect(component.find(GroupedItemList).length).toBe(0);
            expect(component.find(ItemList).length).toBe(1);
        });
    });

    describe('when `state.concern` equals `FILTER', () => {
        let component;

        beforeEach(() => {
            component = shallow(<ItemListPage
                csrf={'3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=='}
                departments={departmentListMock}
                items={itemListMock}
            />);
            component.setState({ concern: FILTER_CONCERN.FILTER });
        });

        afterEach(() => {
            component = null;
        });

        test('it renders an `ItemList` component', () => {
            expect(component.find(ItemList).length).toBe(1);
        });

        test('it renders a `select` element', () => {
            expect(component.find('select').length).toBe(1);
        });
    });
});
