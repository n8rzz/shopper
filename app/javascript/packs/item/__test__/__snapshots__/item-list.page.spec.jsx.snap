// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListPage renders correctly with valid props 1`] = `
<Fragment>
  <FilterBar
    initialSelection="Alpha"
    items={
      Array [
        Object {
          "text": "Alpha",
          "url": "",
        },
        Object {
          "text": "Department",
          "url": "",
        },
        Object {
          "text": "Filter",
          "url": "",
        },
      ]
    }
    onChangeFilterHandler={[Function]}
  />
  <ItemList
    csrf="3yIq75tyV791TFkzSZo/S4b+TVa/N5J3dI14PzMUPviGViEGeDSnGnvNix+MaB0RB0XuV7bpX8I3SmIEAr4/zQ=="
    departmentMap={
      Object {
        "3": Object {
          "created_at": "2019-07-21T16:03:32.883Z",
          "id": 3,
          "name": "Bakery",
          "updated_at": "2019-07-21T16:03:32.883Z",
        },
        "5": Object {
          "created_at": "2019-07-21T16:03:32.883Z",
          "id": 5,
          "name": "Meat",
          "updated_at": "2019-07-21T16:03:32.883Z",
        },
      }
    }
    itemList={
      Array [
        Object {
          "created_at": "2019-07-21T16:03:33.162Z",
          "department_id": 5,
          "id": 15,
          "name": "Agave Syrup",
          "updated_at": "2019-07-21T16:03:33.162Z",
        },
        Object {
          "created_at": "2019-07-21T16:03:33.162Z",
          "department_id": 3,
          "id": 13,
          "name": "Agave Syrup",
          "updated_at": "2019-07-21T16:03:33.162Z",
        },
      ]
    }
  />
</Fragment>
`;
